common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -139,10 +139,309 @@ static bool dwpal_get_channel_scan_freq(const std::vector<unsigned int> &channel
+
+    return true;
+}
+
+static void get_ht_oper(const uint8_t *data, sChannelScanResults &results)
+{
+    if (!data) {
+        LOG(ERROR) << "data buffer is NULL";
+        return;
+    }
+
+    if (!(data[1] & 0x3)) {
+        results.operating_channel_bandwidth = eChannel_Bandwidth_20MHz;
+    } else if ((data[1] & 0x3) != 2) {
+        results.operating_channel_bandwidth = eChannel_Bandwidth_40MHz;
+    }
+
+    results.supported_standards.push_back(eStandard_802_11n);
+    results.operating_standards = eStandard_802_11n;
+}
+
+static void get_vht_oper(const uint8_t *data, sChannelScanResults &results)
+{
+    if (!data) {
+        LOG(ERROR) << "data buffer is NULL";
+        return;
+    }
+
+    if (data[0] == 0x01) {
+        if (data[2]) {
+            results.operating_channel_bandwidth = eChannel_Bandwidth_160MHz;
+        } else {
+            results.operating_channel_bandwidth = eChannel_Bandwidth_80MHz;
+        }
+    }
+
+    if (data[0] == 0x02) {
+        results.operating_channel_bandwidth = eChannel_Bandwidth_80MHz;
+    }
+
+    if (data[0] == 0x03) {
+        results.operating_channel_bandwidth = eChannel_Bandwidth_80_80;
+    }
+
+    if (data[0] > 0x03) {
+        LOG(ERROR) << "illegal TYPE_VHT_OPERATION value=" << data[0];
+    }
+
+    if (results.operating_frequency_band == eOperating_Freq_Band_5GHz) {
+        results.supported_standards.push_back(eStandard_802_11ac);
+        results.operating_standards = eStandard_802_11ac;
+    }
+}
+
+static void get_supprates(const uint8_t *data, uint8_t len, sChannelScanResults &results)
+{
+    uint8_t rate_mbs_fp_8_1;
+    uint32_t rate_kbs;
+
+    if (!data) {
+        LOG(ERROR) << "data buffer is NULL";
+        return;
+    }
+
+    for (int i = 0; i < len; i++) {
+        rate_mbs_fp_8_1 = data[i] & 0x7f;
+
+        if (rate_mbs_fp_8_1 / 2 == 11) {
+            if (results.operating_frequency_band == eOperating_Freq_Band_2_4GHz) {
+                results.supported_standards.push_back(eStandard_802_11b);
+                results.operating_standards = eStandard_802_11b;
+            }
+        } else if (rate_mbs_fp_8_1 / 2 == 54) {
+            if (results.operating_frequency_band == eOperating_Freq_Band_5GHz) {
+                results.supported_standards.push_back(eStandard_802_11a);
+                results.operating_standards = eStandard_802_11a;
+            }
+        }
+
+        /**
+         * rate_mbs_fp_8_1 is tx data rate in mbps 
+         * represented with fixed point u<8,1>.
+         * converting to kbps (no fixpoint) for simplicity u<8,0> 
+         */
+        rate_kbs = (rate_mbs_fp_8_1 / 2) * 1000 + (5 * (rate_mbs_fp_8_1 & 1)) * 100;
+
+        if (data[i] & 0x80) {
+            results.basic_data_transfer_rates_kbps.push_back(rate_kbs);
+        } else {
+            results.supported_data_transfer_rates_kbps.push_back(rate_kbs);
+        }
+    }
+}
+
+static void parse_info_elements(unsigned char *ie, int ielen, sChannelScanResults &results)
```

Comment:
```suggestion
static bool parse_info_elements(const std::string ie, sChannelScanResults &results);
```
Then all the parsing can be c++ style.

Can be done as part of #665